provisioners/windows/geckotwin1064ht-bootstrap.ps1

```diff
@@ -0,0 +1,288 @@
+  )
+  begin {
+    Write-Log -message ('{0} :: begin - {1:o}' -f $($MyInvocation.MyCommand.Name), (Get-Date).ToUniversalTime()) -severity 'DEBUG'
+  }
+  process {
+
+    Set-Location $env:systemdrive\ronin
+    If(!(test-path $logdir\old))  {
+      New-Item -ItemType Directory -Force -Path $logdir\old
+    }
+    If ($stage -eq "inprogress") {
+      git pull https://github.com/$sourceOrg/$sourceRepo $sourceRev
+      $git_exit = $LastExitCode
+      if ($git_exit -eq 0) {
+        $git_hash = (git rev-parse --verify HEAD)
+        Set-ItemProperty -Path HKLM:\SOFTWARE\Mozilla\ronin_puppet -name githash -type  string -value $git_hash
+        Write-Log -message  ('{0} :: Checking/pulling updates from https://github.com/{1}/{2}. Branch: {3}.' -f $($MyInvocation.MyCommand.Name), ($sourceOrg), ($sourceRepo), ($sourceRev)) -severity 'DEBUG'
+      } else {
+        Write-Log -message  ('{0} :: Git pull failed! https://github.com/{1}/{2}. Branch: {3}.' -f $($MyInvocation.MyCommand.Name), ($sourceOrg), ($sourceRepo), ($sourceRev)) -severity 'DEBUG'
+        DO {
+          Start-Sleep -s 60
+          git pull https://github.com/$sourceOrg/$sourceRepo $sourceRev
+          $git_exit = $LastExitCode
+        } Until ( $git_exit -eq 0)
+      }
+    }
+    Set-ItemProperty -Path "HKLM:\SOFTWARE\Mozilla\ronin_puppet" -Name 'bootstrap_stage' -Value 'inprogress'
+
+    # Setting Env variabes for PuppetFile install and Puppet run
+    # The ssl variables are needed for R10k
+    Write-Log -message  ('{0} :: Setting Puppet enviroment.' -f $($MyInvocation.MyCommand.Name)) -severity 'DEBUG'
+    $env:path = "$env:programfiles\Puppet Labs\Puppet\puppet\bin;$env:programfiles\Puppet Labs\Puppet\bin;$env:path"
+    $env:SSL_CERT_FILE = "$env:programfiles\Puppet Labs\Puppet\puppet\ssl\cert.pem"
+    $env:SSL_CERT_DIR = "$env:programfiles\Puppet Labs\Puppet\puppet\ssl"
+    $env:FACTER_env_windows_installdir = "$env:programfiles\Puppet Labs\Puppet"
+    $env:HOMEPATH = "\Users\Administrator"
+    $env:HOMEDRIVE = "C:"
+    $env:PL_BASEDIR = "$env:programfiles\Puppet Labs\Puppet"
+    $env:PUPPET_DIR = "$env:programfiles\Puppet Labs\Puppet"
+    $env:RUBYLIB = "$env:programfiles\Puppet Labs\Puppet\lib"
+    $env:USERNAME = "Administrator"
+    $env:USERPROFILE = "$env:systemdrive\Users\Administrator"
+
+    Write-Log -message  ('{0} :: Installing Puppetfile .' -f $($MyInvocation.MyCommand.Name)) -severity 'DEBUG'
+    R10k puppetfile install  --moduledir=r10k_modules
+   	# Needs to be removed from path or a wrong puppet file will be used
+    $env:path = ($env:path.Split(';') | Where-Object { $_ -ne "$env:programfiles\Puppet Labs\Puppet\puppet\bin" }) -join ';'
+    Get-ChildItem -Path $logdir\*.log -Recurse | Move-Item -Destination $logdir\old -ErrorAction SilentlyContinue
+    puppet apply manifests\nodes\nodes.pp --test --modulepath=modules`;r10k_modules --hiera_config=hiera.yaml --logdest $logdir\$datetime-bootstrap-puppet.log
```

Comment:
Instead of manifests\nodes\nodes.pp as an entry point, I would suggest just 'manifests'.  That allows site.pp to be included and ensures there is no default node.  Otherwise, there will be no toplevel scope. I would also remove --test and be explicit with what --test implies. eg --verbose, etc.